src/coldcard.js

```diff
@@ -507,107 +516,177 @@ export class ColdcardSignMultisigTransaction extends ColdcardInteraction {
-    if (typeof jsonConfig === "object") {
-      this.jsonConfig = jsonConfig;
-    } else if (typeof jsonConfig === "string") {
-      try {
-        this.jsonConfig = JSON.parse(jsonConfig);
-      } catch (error) {
-        throw new Error("Unable to parse JSON.");
-      }
-    } else {
-      throw new Error("Not valid JSON.");
-    }
-
-    if ((this.jsonConfig.uuid || this.jsonConfig.name)) {
-      this.name = this.jsonConfig.uuid || this.jsonConfig.name;
-    } else {
-      throw new Error("Configuration file needs a UUID or a name.");
-    }
-
-    if (this.jsonConfig.quorum.requiredSigners && this.jsonConfig.quorum.totalSigners) {
-      this.requiredSigners = this.jsonConfig.quorum.requiredSigners;
-      this.totalSigners = this.jsonConfig.quorum.totalSigners;
-    } else {
-      throw new Error("Configuration file needs quorum.requiredSigners and quorum.totalSigners.");
-    }
-
-    if (this.jsonConfig.addressType) {
-      this.addressType = jsonConfig.addressType;
-    } else {
-      throw new Error("Configuration file needs addressType.");
+   * @description Takes a coldcard config string and creates
+   * a new instance of a MultisigWalletConfig.
+   * 
+   * NOTE for ColdCard: only the root xfp of the signing device is required to be correct, but we
+   * recommend only setting up the multisig wallet on the Coldcard with complete xfp
+   * information.
+   * 
+   * This is an example Coldcard config file from
+   * https://coldcardwallet.com/docs/multisig
+   * 
+   * # Coldcard Multisig setup file (exported from 4369050F)
+   * #
+   * Name: MeMyself
+   * Policy: 2 of 4
+   * Derivation: m/45'
+   * Format: P2WSH
+   * 
+   * D0CFA66B: tpubD9429UXFGCTKJ9NdiNK4rC5...DdP9
+   * 8E697B74: tpubD97nVL37v5tWyMf9ofh5rzn...XgSc
+   * BE26B07B: tpubD9ArfXowvGHnuECKdGXVKDM...FxPa
+   * 4369050F: tpubD8NXmKsmWp3a3DXhbihAYbY...9C8n
+   * 
+   * @param {string} configString valid coldcard config string
+   * @returns {MultisigWalletConfig} new instance of MultisigWalletConfig
+   * @example
+   * import { parseColdcardConfig } from "unchained-wallets";
+   * 
+   * const config = MultisigWalletConfig.fromColdCardConfig(ccFileData);
+   * console.log(JSON.parse(config.toJSONConfig()))
+   * 
+   * {
+   *    name: MeMyself,
+   *    quorum: {
+   *      totalSigners: 4,
+   *      requiredSigners: 2,   
+   *    },
+   *    addressType: 'P2WSH',
+   *    network: 'testnet',
+   *    extendedPublicKeys: [
+   *      {
+   *        xfp: 'D0CFA66B',
+   *        xpub: 'tpubD9429UXFGCTKJ9NdiNK4rC5...DdP9'
+   *      },
+   *      {
+   *        xfp: '8E697B74',
+   *        xpub: 'tpubD97nVL37v5tWyMf9ofh5rzn...XgSc',
+   *      },
+   *      {
+   *        xfp: 'BE26B07B',
+   *        xpub: 'tpubD9ArfXowvGHnuECKdGXVKDM...FxPa',
+   *      },
+   *      {
+   *        xfp: '4369050F',
+   *        xpub: 'tpubD8NXmKsmWp3a3DXhbihAYbY...9C8n',
+   *      }
+   *    ]
+   * }
+   */
+export function parseColdcardConfig(configString) {
```

Comment:
i didnt compare line by line but this would be good to compare how they do it vs how you've done it
https://github.com/Coldcard/firmware/blob/master/shared/multisig.py#L523

Reply:
Ah, good reference. Thanks!